### ISSUE TITLE
Unintended indent for multiline template string argument combined with another multiline argument

### ISSUE THREAD

**Issue by markatom** [eslint/eslint#9061]
**Tell us about your environment**

* **ESLint Version:** 4.3.0
* **Node Version:** 8.1.3
* **npm Version:** 5.0.3

**What parser (default, Babel-ESLint, etc.) are you using?**

I tried both default and babel-eslint. They're giving me same result.

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

{
    "env": {
        "es6": true,
        "node": true
    },
    "extends": [
        "eslint:recommended"
    ],
    "rules": {
        "indent": ["error", 4]
    }
}

```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js

const f = (first, second) => {
    return second;
};

// works for multiline template strings
f(`
    multiline
    template
    string
`, `
    multiline
    template
    string
`);

// works for multiline objects
f({
    foo: 'foo',
    bar: 'bar',
}, {
    foo: 'foo',
    bar: 'bar',
});

// doesn't work for multiline template string and multiline object
f(`
    multiline
    template
    string
`, {
    foo: 'foo',
    bar: 'bar',
});


```

**What did you expect to happen?**

I expected all of the three examples will pass.

**What actually happened? Please include the actual, raw output from ESLint.**

```
  31:1  error  Expected indentation of 8 spaces but found 4  indent
  32:1  error  Expected indentation of 8 spaces but found 4  indent
  33:1  error  Expected indentation of 4 spaces but found 0  indent
```

I believe the eslint fix is incorrect

```diff
 const f = (first, second) => {
     return second;
 };
 
 // works for multiline template strings
 f(`
     multiline
     template
     string
 `, `
     multiline
     template
     string
 `);

 // works for multiline objects
 f({
     foo: 'foo',
     bar: 'bar',
 }, {
     foo: 'foo',
     bar: 'bar',
 });

 // doesn't work for multiline template string and multiline object
 f(`
     multiline
     template
     string
 `, {
-    foo: 'foo',
-    bar: 'bar',
-});
+        foo: 'foo',
+        bar: 'bar',
+    });
```

Could you please confirm this is an unintended behaviour? Thank you.

**Comment by not-an-aardvark** [comment]
We actually have a [test](https://github.com/eslint/eslint/blob/37158c58eef18ff6d61a9e6ffdf62fe7dc5a03af/tests/lib/rules/indent.js#L3994-L4005) asserting the current behavior, so I think this is working as intended. I don't recall if there was a specific reason why I added that test, though. I think I agree that your proposed behavior makes more sense.

**Comment by markatom** [comment]
Thank you for a quick response. Can I expect the behaviour will be changed then?

**Comment by markatom** [comment]
Any updates on this issue?

**Comment by platinumazure** [comment]
@not-an-aardvark Where are we at on this issue?

**Comment by not-an-aardvark** [comment]
Marked as accepted.

**Comment by gurpreetatwal** [comment]
@not-an-aardvark anything we can do to help get this moving forward?

**Comment by not-an-aardvark** [comment]
You could submit a PR.

**Comment by gurpreetatwal** [comment]
@not-an-aardvark thank you! I tried working on it, but was still trying to understand how eslint works haha
